### PR TITLE
Apps API

### PR DESCRIPTION
**Pull request by adamruzicka**
Follows:
- [x] pundit PR https://github.com/RedHatInsights/notifications-backend/pull/6

TODO:
- [x] describe `apps#show`
- [x] controller, policy, specs for endpoints
- [x] controller, policy, specs for filters

### CODE REVIEW

**Comment by ShimShtein** [comment]
@adamruzicka, this needs a rebase.

File: spec/integration/apps_spec.rb [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    after { App.destroy_all }
+    get 'List all apps' do
+      tags 'app'
+      description 'Lists all apps requested'
+      consumes 'application/json'
+      produces 'application/json'
+      parameter name: :'X-RH-IDENTITY', in: :header, schema: { type: :string }
+
+      response '200', 'lists all apps requested' do
+        let(:blog) { { title: 'foo', content: 'bar' } }
```
Comment (ShimShtein) [inline spec/integration/apps_spec.rb]: Leftover?

File: spec/integration/apps_spec.rb [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+        let(:'X-RH-IDENTITY') { encoded_header }
+        schema type: :object,
+               properties: {
+                 # meta: { '$ref' => '#/definitions/metadata' },
+                 # links: { '$ref' => '#/definitions/links' },
```
Comment (ShimShtein) [inline spec/integration/apps_spec.rb]: This and previous line look like a leftover

**Comment by adamruzicka** [comment]
The addition on `capybara-webkit` broke `bundle install` in jenkins as there are unsatisfied dependencies. This will need to be sorted in another PR which will have to be merged before this one

**Comment by ShimShtein** [comment]
@adamruzicka, https://github.com/RedHatInsights/notifications-backend/pull/11 is merged, can you rebase and kick off jenkins again?

**Comment by ShimShtein** [comment]
Is it still WIP? Or should I test and merge it?

**Comment by adamruzicka** [comment]
The changes inside this PR should be good to go, unless we want to implement something more as part of this

**Comment by ShimShtein** [comment]
That's a good start. I'll test and merge it on Sunday.

File: spec/integration/apps_spec.rb [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+require 'swagger_helper'
+
+# Justification: It's mostly hash test data
+# rubocop:disable Metrics/MethodLength
+def encoded_header
+  Base64.encode64(
+    {
+      'identity':
+      {
+        'account_number': '1234',
```
Comment (ShimShtein) [inline spec/integration/apps_spec.rb]: Accounts don't have a unique index on `account_number`

File: Gemfile [hunk 4]
```diff
@@ -35,10 +35,12 @@ gem 'rswag-ui'
 gem 'fast_jsonapi'
 
 gem 'pundit'
+gem 'will_paginate'
 
 group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
   gem 'byebug', platforms: %i[mri mingw x64_mingw]
+  gem 'capybara-webkit'
```
[the remark this answers is not in the document]
Comment (adamruzicka) [reply Gemfile]: Forgot to push the commit actually removing things

**Comment by ShimShtein** [comment]
Took the liberty to address insignificant rubocop comments and add swagger generation to our containers.

Thanks @adamruzicka!

**Pull request merged**